packages/@aws-cdk/aws-ecs/lib/base/task-definition.ts

```diff
@@ -293,6 +293,39 @@ export class TaskDefinition extends TaskDefinitionBase {
     return this._executionRole;
   }
 
+  /**
+   * Returns the host port that match the provided container name and container port.
+   *
+   * @internal
+   */
+  public _findHostPort(target: LoadBalancerTargetOptions): HostPort {
+    const container = this.findContainer(target.containerName);
+    if (container === undefined) {
+      throw new Error("Container does not exist.");
+    }
+    const portMapping = container._findPortMapping(target.containerPort, target.protocol || Protocol.TCP);
+    if (portMapping === undefined) {
+      throw new Error("Container port using the protocol does not exist.");
+    }
+    if (portMapping.hostPort !== undefined && portMapping.hostPort !== 0) {
+      return {
+        port: portMapping.hostPort,
+        protocol: portMapping.protocol || Protocol.TCP
+      };
+    }
+
+    if (this.networkMode === NetworkMode.BRIDGE) {
```

Comment:
Is the order of `if`s here correct? Do we really check port mappings before checking network mode?

Reply:
Not sure if I fully understand what you mean. Is there any problem with the logic: check if port mapping exists then determine the security group port to open?

Reply:
If the user somehow specifies a host port in bridge mode (and this is a public function so they could call it directly with too much information) I believe the early return would cause an incorrect binding to that port instead of the ephemeral range. Moving this above the port check would fix that edge case.

Reply:
But the user specifies a host port in bridge mode, it should return the host port right? I think it returns the ephemeral range only if the network mode is bridge and the host port is 0 (or undefined). See [hostPort](https://docs.aws.amazon.com/AmazonECS/latest/APIReference/API_PortMapping.html).

Reply:
You're right, disregard this one :)

Reply:
Just trying to be sure. This looks and feels like hairy logic.